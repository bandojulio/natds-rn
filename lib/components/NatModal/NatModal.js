import React from 'react';
import PropTypes from 'prop-types';

import {
  ModalView,
  ModalText,
  ModalTitle,
  ModalButton,
  ModalButtonLabel,
} from './NatModal.styles';

export const NatModal = ({
  title,
  text,
  buttonText,
  closeAction,
}) => (
  <ModalView>
    <ModalTitle>{title}</ModalTitle>
    <ModalText>{text}</ModalText>
    <ModalButton onPress={closeAction}>
      <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>
    </ModalButton>
  </ModalView>
);

NatModal.propTypes = {
  title: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  buttonText: PropTypes.string.isRequired,
  closeAction: PropTypes.func.isRequired,
};
