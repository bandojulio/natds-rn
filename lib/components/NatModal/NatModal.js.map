{"version":3,"sources":["NatModal.js"],"names":["NatModal","title","text","action","buttonText","alternative","buttonTextAlternative","alternativeAction","save","toUpperCase","propTypes","PropTypes","func","isRequired","bool","string","defaultProps"],"mappings":"2IAAA,4B;AACA;AACA,qC;;AAEA,2C;;;;;;;;;;;;AAYO,GAAMA,2BAAW,QAAXA,SAAW;AACtBC,KADsB,MACtBA,KADsB;AAEtBC,IAFsB,MAEtBA,IAFsB;AAGtBC,MAHsB,MAGtBA,MAHsB;AAItBC,UAJsB,MAItBA,UAJsB;AAKtBC,WALsB,MAKtBA,WALsB;AAMtBC,qBANsB,MAMtBA,qBANsB;AAOtBC,iBAPsB,MAOtBA,iBAPsB;AAQtBC,IARsB,MAQtBA,IARsB;;AAUtB,8BAAC,kBAAD;AACE,cAAc,MADhB;AAEE,gBAFF;AAGE,eAAgBL,MAHlB;AAIE,KAAMK,IAJR;;AAME,8BAAC,mBAAD;AACE,8BAAC,sBAAD,EAAc,mBAAoBP,KAAlC;AACE,8BAAC,oBAAD,EAAY,KAAMO,IAAlB,iDAAyBP,KAAzB,CADF;AAEI,CAACO,IAAD,EAAU,8BAAC,mBAAD,kDAAYN,IAAZ,CAFd;;AAIMG;;AAEI,8BAAC,wBAAD;AACE,8BAAC,gCAAD;AACE,YAAaA,WADf;AAEE,QAASE,iBAFX;AAGE,mBAAoBD,sBAAsBG,WAAtB,EAHtB;;AAKE,8BAAC,qCAAD;AACE,YAAaJ,WADf;;AAGGC,sBAAsBG,WAAtB,EAHH,CALF,CADF;;;;AAaE,8BAAC,qBAAD;AACE,YAAaJ,WADf;AAEE,QAASF,MAFX;AAGE,mBAAoBC,WAAWK,WAAX,EAHtB;;AAKE,8BAAC,0BAAD,kDAAmBL,WAAWK,WAAX,EAAnB,CALF,CAbF,CAFJ;;;;;AAyBI,8BAAC,qBAAD;AACE,YAAaJ,WADf;AAEE,QAASF,MAFX;AAGE,mBAAoBC,WAAWK,WAAX,EAHtB;;AAKE,8BAAC,0BAAD,kDAAmBL,WAAWK,WAAX,EAAnB,CALF,CA7BV,CADF,CANF,CAVsB,GAAjB;;;;;;;;;AA4DPT,SAASU,SAAT,CAAqB;AACnBP,OAAQQ,oBAAUC,IAAV,CAAeC,UADJ;AAEnBR,YAAaM,oBAAUG,IAFJ;AAGnBP,kBAAmBI,oBAAUC,IAHV;AAInBR,WAAYO,oBAAUI,MAAV,CAAiBF,UAJV;AAKnBP,sBAAuBK,oBAAUI,MALd;AAMnBP,KAAMG,oBAAUG,IANG;AAOnBZ,KAAMS,oBAAUI,MAPG;AAQnBd,MAAOU,oBAAUI,MAAV,CAAiBF,UARL,CAArB;;;AAWAb,SAASgB,YAAT,CAAwB;AACtBX,YAAa,KADS;AAEtBE,kBAAmBI,oBAAUC,IAFP;AAGtBN,sBAAuB,mBAHD;AAItBE,KAAM,KAJgB;AAKtBN,KAAM,kBALgB,CAAxB","file":"NatModal.js","sourcesContent":["import React from 'react';\nimport { Modal } from 'react-native';\nimport PropTypes from 'prop-types';\n\nimport {\n  ModalView,\n  ModalContent,\n  ModalText,\n  ModalTitle,\n  ModalButton,\n  ModalButtonLabel,\n  ModalContainer,\n  ModalButtonAlternative,\n  ModalButtonLabelAlternative,\n} from './NatModal.styles';\n\nexport const NatModal = ({\n  title,\n  text,\n  action,\n  buttonText,\n  alternative,\n  buttonTextAlternative,\n  alternativeAction,\n  save,\n}) => (\n  <Modal\n    animationType=\"fade\"\n    transparent\n    onRequestClose={action}\n    save={save}\n  >\n    <ModalView>\n      <ModalContent accessibilityLabel={title}>\n        <ModalTitle save={save}>{title}</ModalTitle>\n        { !save && (<ModalText>{text}</ModalText>) }\n        {\n            alternative\n              ? (\n                <ModalContainer>\n                  <ModalButtonAlternative\n                    alternative={alternative}\n                    onPress={alternativeAction}\n                    accessibilityLabel={buttonTextAlternative.toUpperCase()}\n                  >\n                    <ModalButtonLabelAlternative\n                      alternative={alternative}\n                    >\n                      {buttonTextAlternative.toUpperCase()}\n                    </ModalButtonLabelAlternative>\n                  </ModalButtonAlternative>\n\n                  <ModalButton\n                    alternative={alternative}\n                    onPress={action}\n                    accessibilityLabel={buttonText.toUpperCase()}\n                  >\n                    <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>\n                  </ModalButton>\n                </ModalContainer>\n              )\n              : (\n                <ModalButton\n                  alternative={alternative}\n                  onPress={action}\n                  accessibilityLabel={buttonText.toUpperCase()}\n                >\n                  <ModalButtonLabel>{buttonText.toUpperCase()}</ModalButtonLabel>\n                </ModalButton>\n              )\n          }\n      </ModalContent>\n    </ModalView>\n  </Modal>\n);\n\nNatModal.propTypes = {\n  action: PropTypes.func.isRequired,\n  alternative: PropTypes.bool,\n  alternativeAction: PropTypes.func,\n  buttonText: PropTypes.string.isRequired,\n  buttonTextAlternative: PropTypes.string,\n  save: PropTypes.bool,\n  text: PropTypes.string,\n  title: PropTypes.string.isRequired,\n};\n\nNatModal.defaultProps = {\n  alternative: false,\n  alternativeAction: PropTypes.func,\n  buttonTextAlternative: 'Button text error',\n  save: false,\n  text: 'Modal text error',\n};\n"]}