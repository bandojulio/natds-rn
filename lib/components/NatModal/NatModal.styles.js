import styled from 'styled-components/native';

import { natura, roboto } from '../../assets/theme/fonts/fonts';
import { colors } from '../../assets/theme/colors/colors';

export const ModalView = styled.View`
  align-items: stretch;
  justify-content: center;
  background-color: ${colors.primary.white};
  border-radius: 6;
  margin-horizontal: 24;
  box-shadow: 0px 16px 32px rgba(0, 0, 0, 0.24);
  elevation: 6;
`;

export const ModalText = styled.Text`
  padding-horizontal: 16;
  margin-bottom: 24;
  font-size: 14;
  color: ${colors.bw.grayDarker};
`;

export const ModalTitle = styled.Text`
  color: ${colors.bw.black};
  font-size: 20;
  font-family: ${natura.bold}
  margin-top: 24;
  margin-bottom: 16;
  padding-horizontal: 16;
  align-self: flex-start;
`;

export const ModalButton = styled.TouchableOpacity`
  align-items: center
  padding: 16px;
  background-color: ${colors.primary.yellow}
  border-bottom-left-radius: 6
  border-bottom-right-radius: 6
`;

export const ModalButtonLabel = styled.Text`
  color: ${colors.primary.white}
  font-family: ${roboto.bold};
  letter-spacing: 1px
`;
