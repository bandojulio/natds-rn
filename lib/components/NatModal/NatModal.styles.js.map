{"version":3,"sources":["NatModal.styles.js"],"names":["ModalView","styled","View","ModalContent","colors","primary","white","ModalText","Text","Platform","OS","Small","bw","grayDarker","ModalTitle","black","header","XSmallBold","props","save","ModalButton","TouchableOpacity","yellow","alternative","css","ModalButtonLabel","sanfrancisco","bold","roboto","ModalContainer","ModalButtonAlternative","ModalButtonLabelAlternative"],"mappings":"wkFAAA;AACA,gD;;AAEA;AACA;AACA,wD;;AAEO,GAAMA,6BAAYC,iBAAOC,IAAnB,iBAAN;;;;;;;AAOA,GAAMC,mCAAeF,iBAAOC,IAAtB;AACSE,eAAOC,OAAP,CAAeC,KADxB,CAAN;;;;;;;AAQA,GAAMC,6BAAYN,iBAAOO,IAAnB;;;AAGT,qBAAKC,sBAASC,EAAd,EAAkBC,KAHT;AAIFP,eAAOQ,EAAP,CAAUC,UAJR,CAAN;;;AAOA,GAAMC,+BAAab,iBAAOO,IAApB;AACFJ,eAAOQ,EAAP,CAAUG,KADR;AAETC,mBAAOC,UAFE;AAGG,sBAAUC,OAAMC,IAAN,CAAa,EAAb,CAAkB,EAA5B,EAHH;AAIM,sBAAUD,OAAMC,IAAN,CAAa,EAAb,CAAkB,CAA5B,EAJN,CAAN;;;;AAQA,GAAMC,iCAAcnB,iBAAOoB,gBAArB;;;AAGSjB,eAAOC,OAAP,CAAeiB,MAHxB;;;AAMP,sBAASJ,OAAMK,WAAN,KAAqBC,WAArB,mBAAT,EANO,CAAN;;;;;;;AAaA,GAAMC,2CAAmBxB,iBAAOO,IAA1B;AACFJ,eAAOC,OAAP,CAAeC,KADb;AAEKG,sBAASC,EAAT,GAAgB,KAAjB,CAA0BgB,oBAAaC,IAAvC,CAA8CC,cAAOD,IAFzD,CAAN;;;;AAMA,GAAME,uCAAiB5B,iBAAOC,IAAxB,kBAAN;;;;AAIA,GAAM4B,uDAAyB7B,iBAAOoB,gBAAhC;;;;AAISjB,eAAOC,OAAP,CAAeC,KAJxB,CAAN;;;;AAQA,GAAMyB,iEAA8B9B,iBAAOO,IAArC;AACFJ,eAAOC,OAAP,CAAeiB,MADb;AAEKb,sBAASC,EAAT,GAAgB,KAAjB,CAA0BgB,oBAAaC,IAAvC,CAA8CC,cAAOD,IAFzD,CAAN","file":"NatModal.styles.js","sourcesContent":["import { Platform } from 'react-native';\nimport styled, { css } from 'styled-components/native';\n\nimport { roboto, sanfrancisco } from '../../assets/theme/fonts/fonts';\nimport { body, header } from '../../assets/theme/typography/typography';\nimport { colors } from '../../assets/theme/colors/colors';\n\nexport const ModalView = styled.View`\n  flex: 1;\n  justify-content: center;\n  align-items: center;\n  background-color: rgba(3, 3, 3, 0.5);\n`;\n\nexport const ModalContent = styled.View`\n  background-color: ${colors.primary.white};\n  border-radius: 6;\n  box-shadow: 0px 16px 32px rgba(0, 0, 0, 0.24);\n  elevation: 8;\n  margin-horizontal: 24;\n`;\n\nexport const ModalText = styled.Text`\n  padding-horizontal: 16;\n  margin-bottom: 24;\n  ${body(Platform.OS).Small}\n  color: ${colors.bw.grayDarker};\n`;\n\nexport const ModalTitle = styled.Text`\n  color: ${colors.bw.black};\n  ${header.XSmallBold};\n  margin-top: ${props => (props.save ? 40 : 24)};\n  margin-bottom: ${props => (props.save ? 40 : 8)};\n  padding-horizontal: 16;\n`;\n\nexport const ModalButton = styled.TouchableOpacity`\n  align-items: center;\n  padding: 16px;\n  background-color: ${colors.primary.yellow};\n  border-bottom-left-radius: 6;\n  border-bottom-right-radius: 6;\n    ${props => props.alternative && css`\n    width: 50%;\n    border-bottom-left-radius: 0;\n    border-bottom-right-radius: 6;\n`}\n`;\n\nexport const ModalButtonLabel = styled.Text`\n  color: ${colors.primary.white}\n  font-family: ${(Platform.OS === 'ios') ? sanfrancisco.bold : roboto.bold};\n  letter-spacing: 1px;\n`;\n\nexport const ModalContainer = styled.View`\n  flex-direction: row;\n`;\n\nexport const ModalButtonAlternative = styled.TouchableOpacity`\n  width: 50%;\n  align-items: center;\n  padding: 16px;\n  background-color: ${colors.primary.white};\n  border-bottom-left-radius: 6;\n`;\n\nexport const ModalButtonLabelAlternative = styled.Text`\n  color: ${colors.primary.yellow};\n  font-family: ${(Platform.OS === 'ios') ? sanfrancisco.bold : roboto.bold};\n  letter-spacing: 1px;\n`;\n"]}