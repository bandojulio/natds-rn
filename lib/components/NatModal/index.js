Object.defineProperty(exports,"__esModule",{value:true});var _NatModal=require('./NatModal');Object.defineProperty(exports,'NatModal',{enumerable:true,get:function get(){return _NatModal.NatModal;}});
//# sourceMappingURL=index.js.map