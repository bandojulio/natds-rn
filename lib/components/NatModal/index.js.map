{"version":3,"sources":["index.js"],"names":["NatModal"],"mappings":"2LAASA,Q","file":"index.js","sourcesContent":["export { NatModal } from './NatModal';\n"]}