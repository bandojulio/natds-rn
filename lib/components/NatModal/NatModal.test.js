import React from 'react';
import renderer from 'react-test-renderer';

import { NatModal } from './NatModal';

describe('NatModal', () => {
  it('Should match snapshot', () => {
    const props = {
      title: 'Boas vindas!',
      text: 'Texto de boas vindas',
      closeAction: jest.fn(),
      buttonText: 'Entendi',
      visible: true,
    };

    const modal = renderer.create(<NatModal {...props} />).toJSON();

    expect(modal).toMatchSnapshot();
  });
});
