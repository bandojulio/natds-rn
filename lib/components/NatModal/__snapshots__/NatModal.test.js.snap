// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NatModal Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "stretch",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 6,
        "elevation": 6,
        "justifyContent": "center",
        "marginHorizontal": 24,
        "shadowColor": "rgba(0, 0, 0, 0.24)",
        "shadowOffset": Object {
          "height": 16,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 32,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "color": "#000000",
          "fontFamily": "NaturaSans-Bold",
          "fontSize": 20,
          "marginBottom": 16,
          "marginTop": 24,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    Boas vindas!
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#333333",
          "fontSize": 14,
          "marginBottom": 24,
          "paddingHorizontal": 16,
        },
      ]
    }
  >
    Texto de boas vindas
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#F4AB34",
        "borderBottomLeftRadius": 6,
        "borderBottomRightRadius": 6,
        "opacity": 1,
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontFamily": "Roboto-Bold",
            "letterSpacing": 1,
          },
        ]
      }
    >
      ENTENDI
    </Text>
  </View>
</View>
`;
